lib/transaction.rb

```diff
@@ -0,0 +1,41 @@
+require_relative 'tracker'
+require 'date'
+
+class Transaction
+  attr_reader :date, :amount
+
+  def initialize(date, amount)
+    check_if_valid(date, amount)
+    @date = date
+    @amount = amount
+  end
+
+  def format_for_statement
+    @date + ' || ' + format_amount_for_statement
+  end
+
+  private
+
+  def check_if_valid(date, amount)
```

Comment:
Is this method hard to read? It's doing quite a lot.  What would make it easier to read?